filemonitor/FileMonitor.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef FILE_MONITOR_HPP
+#define FILE_MONITOR_HPP
+
+#include "IFileMonitor.hpp"
+#include <iostream>
+#include <fstream>
+#include <filesystem>
+#include <chrono>
+#include <thread>
+
+namespace fs = std::filesystem;
+
+class FileMonitor : public IFileMonitor {
+    public:
+        FileMonitor(const std::string& filePath) : filePath(filePath) {}
+
+        void startMonitoring() override {
+        // get last modification time
+        auto lastModified = fs::last_write_time(filePath);
+
+        while (true) {
+            // wait till next modification time check
+            std::this_thread::sleep_for(std::chrono::milliseconds(500)); 
```

Comment:
it would be great to move 500ms to some constant with appropriate name and not use magic numbers

Reply:
macro added